File: backend/app/schemas/backup.py
from pydantic import BaseModel
from typing import List, Optional, Any
from datetime import datetime

class BackupTaskSchema(BaseModel):
    id: Optional[str] = None
    name: str
    mode: str # '7z', 'tar', 'sync'
    src_path: str
    dst_path: str
    password: Optional[str] = None
    enabled: bool = True
    schedule_type: str # 'cron', 'interval'
    schedule_value: str
    ignore_patterns: Optional[List[str]] = []
    
    # 新增字段
    storage_type: str = "ssd" # 'ssd', 'hdd', 'cloud'
    compression_level: int = 1 # 1-9
    sync_strategy: str = "mirror" # 'mirror', 'incremental'

class BackupHistorySchema(BaseModel):
    id: int
    task_id: str
    task_name: str
    start_time: datetime
    end_time: Optional[datetime]
    status: str
    mode: str
    size: float
    message: Optional[str]
    output_path: Optional[str]

    class Config:
        from_attributes = True

File: frontend/src/api/backup.ts
import request from '@/utils/request'

export interface BackupTask {
  id?: string
  name: string
  mode: string
  storage_type: string
  sync_strategy: string
  compression_level: number
  src_path: string
  dst_path: string
  password?: string
  enabled: boolean
  schedule_type: string
  schedule_value: string
  ignore_patterns: string[]
  host_id: string
}

export const backupApi = {
  getTasks: () => request.get<BackupTask[]>('/api/backup/tasks'),
  saveTask: (task: BackupTask) => {
    if (task.id) {
      return request.put(`/api/backup/tasks/${task.id}`, task)
    } else {
      return request.post('/api/backup/tasks', task)
    }
  },
  runTask: (id: string) => request.post(`/api/backup/tasks/${id}/run`),
  getHistory: (taskId?: string) => request.get('/api/backup/history', { params: { task_id: taskId } })
}